src/App.tsx

```diff
@@ -35,41 +33,44 @@ export const App = () => {
         let routes: Route[] = []
 
         // keep track of what's currently registered
-        const currentChainAdapters = chainAdapterManager
-          .getSupportedAdapters()
-          .map(adapter => adapter().getType())
+        const currentChainAdapters = chainAdapterManager.getSupportedChains()
 
         // newly registered will be default + what comes from plugins
-        const newChainAdapters: ChainTypes[] = Object.keys(defaultUnchainedUrls) as ChainTypes[]
+        const newChainAdapters: { [k in ChainTypes]?: () => ChainAdapter<ChainTypes> } = {}
 
         // register providers from each plugin
         for (const [, plugin] of pluginManager.entries()) {
           // Ignore plugins that have their feature flag disabled
           // If no featureFlag is present, then we assume it's enabled
           if (!plugin.featureFlag || featureFlags[plugin.featureFlag]) {
             // routes providers
-            routes = routes.concat(plugin.routes)
+            if (plugin.routes) {
+              routes = routes.concat(plugin.routes)
+            }
 
             // chain adapters providers
             plugin.providers?.chainAdapters?.forEach(([chain, factory]) => {
-              chainAdapterManager.addChain(chain, factory)
               // track newly registered adapters by plugins
-              newChainAdapters.push(chain)
+              newChainAdapters[chain] = factory
```

Comment:
please check with switching wallets, this is not working correctly w/ metamask. i know metamask doesnt support cosmos, but the dashboard doesn't load when enabling the flag

Reply:
the fix was in `getAccountSpecifiers`. We definitely need to move this into redux state.

Reply:
Switched back and forth between Metamask and native wallet with this diff applied, this seems fixed